### PR TITLE
INS-19101 cadence cdc resize via terraform

### PR DESCRIPTION
**Pull request by lancezhao-ins**
Internal reference - INS-19376
Internal reference - INS-19101
Internal reference - INS-19451

### CODE REVIEW

File: instaclustr/resource_cluster_unit_test.go [hunk 1]
```diff
@@ -501,11 +501,11 @@ func TestHasCassandraSizeChanges(t *testing.T) {
 func TestDoClusterResizeDefault(t *testing.T) {
 	err := doClusterResize(MockApiClient{
 		cluster: Cluster{
-			ID:         "CADENCE",
-			BundleType: "CADENCE",
+			ID:         "POSTGRESQL",
```
Comment (WeiXinFam-Insta) [inline instaclustr/resource_cluster_unit_test.go:504]: **[QUESTION]** Why did you change the BundleType from `Cadence` to `PostgreSQL`?

**Review by WeiXinFam-Insta — COMMENTED** [review]
Left some questions, also since you've added a new field `use_cadence_web_auth`, wouldn't it be better if you add some tests including this field and also update the `main.tf` under the examples folder?

File: instaclustr/resource_cluster.go [hunk 2]
```diff
@@ -614,6 +614,11 @@ func resourceCluster() *schema.Resource {
 										Optional: true,
 										ForceNew: true,
 									},
+									"use_cadence_web_auth": {
+                                        Type:     schema.TypeBool,
+                                        Optional: true,
```
[the remark this answers is not in the document]
Comment (lancezhao-ins) [reply instaclustr/resource_cluster.go:619]: weird, looks good locally.

File: instaclustr/resource_cluster.go [hunk 3]
```diff
@@ -1143,6 +1152,13 @@ func doClusterResize(client APIClientInterface, clusterID string, d resourceData
 		} else {
 			return nil
 		}
+	case "CADENCE":
+		if hasCadenceSizeChanges(d) {
```
Comment (arjunrajinstaclustr) [inline instaclustr/resource_cluster.go:1156]: Looks like `hasCassandraSizeChanges`, `hasRedisSizeChanges` and `hasCadenceSizeChanges` do the exact same thing. How about we merge them up into a single method? Example - `hasSimpleNodeSizeChanges`

File: instaclustr/resource_cluster_unit_test.go (hunk 1 above)
Comment (lancezhao-ins) [reply instaclustr/resource_cluster_unit_test.go:504]: It is test the default behavior when users perform a resize on a NOT supported bundle type. Before is Cadence, now change to POSTGRESQL.

File: instaclustr/resource_cluster.go (hunk 3 above)
Comment (lancezhao-ins) [reply instaclustr/resource_cluster.go:1156]: Good idea, on it.

**Review by matthewmillett-instaclustr — APPROVED** [review]
LGTM

File: instaclustr/resource_cluster.go [hunk 4]
```diff
@@ -1114,7 +1115,7 @@ func doClusterResize(client APIClientInterface, clusterID string, d resourceData
 	}
 	switch cluster.BundleType {
 	case "APACHE_CASSANDRA":
-		if hasCassandraSizeChanges(d) {
+		if hasSimpleNodeSizeChanges(d) {
```
Comment (WeiXinFam-Insta) [inline instaclustr/resource_cluster.go:1118]: **[SUGGESTIONS]** There seems like a lot of repetition of the if else for each bundle. Is there a way to refactor it? 
`if(hasSimpleNodeSizeChanges(d)){return bundleClusterResize}`
The function `bundleClusterResize` takes the bundle type as parameter and do the switch check there.
Not quite sure the best way to go about it but that's one of the ideas^^

**Review by WeiXinFam-Insta — APPROVED** [review]
Just some thoughts on code refactoring otherwise all g

**Review by arjunrajinstaclustr — DISMISSED** [review]

**Review by arjunrajinstaclustr — APPROVED** [review]

**Comment by peter-toms-instaclustr** [comment]
rock and roll

**Review by peter-toms-instaclustr — APPROVED** [review]
